pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/utils/SchemaValidationFunctions.kt

```diff
@@ -0,0 +1,399 @@
+package gov.cdc.ocio.processingstatusapi.utils
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.JsonNodeType
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+
+import com.networknt.schema.JsonSchema
+import com.networknt.schema.JsonSchemaFactory
+import com.networknt.schema.SpecVersion
+import com.networknt.schema.ValidationMessage
+import gov.cdc.ocio.processingstatusapi.ReportManager
+import gov.cdc.ocio.processingstatusapi.exceptions.BadRequestException
+import gov.cdc.ocio.processingstatusapi.models.reports.CreateReportSBMessage
+import gov.cdc.ocio.processingstatusapi.models.reports.MessageMetadata
+import gov.cdc.ocio.processingstatusapi.models.reports.Source
+import gov.cdc.ocio.processingstatusapi.models.reports.StageInfo
+import gov.cdc.ocio.processingstatusapi.utils.SchemaValidation.Companion.logger
+import gov.cdc.ocio.processingstatusapi.utils.SchemaValidation.Companion.reason
+
+import java.awt.datatransfer.MimeTypeParseException
+import java.io.File
+import java.time.Instant
+import java.util.*
+import javax.activation.MimeType
+
+/**
+ * Function checks if input string is a valid JSON
+ * @param jsonString to be checked
+ * @return boolean true if its valid JSON otherwise false
+ */
+fun isJsonValid(jsonString: String): Boolean {
+    return try {
+        val mapper = jacksonObjectMapper()
+        mapper.readTree(jsonString)
+        true
+    } catch (e: Exception) {
+        false
+    }
+}
+
+/**
+ * Checks for depreciated fields within message that are still accepted for backward compatability.
+ * If any depreciated fields are found, they are replaced with their corresponding new fields.
+ *
+ * @param messageAsString message to be checked against depreciated fields.
+ * @return updated message if depreciated fields were found.
+ */
+fun checkAndReplaceDeprecatedFields(messageAsString: String): String {
```

Comment:
This function might be best if made part of the Report class.  It could be a companion object.